server/lib/src/idm/credupdatesession.rs

```diff
@@ -1848,11 +1847,9 @@ impl<'a> IdmServerCredUpdateTransaction<'a> {
             return Err(OperationError::AccessDenied);
         };
 
-        // Is there something else in progress?
-        // Or should this just cancel it ....
+        // Is there something else in progress? Cancel it if so.
         if !matches!(session.mfaregstate, MfaRegState::None) {
-            admin_info!("Invalid TOTP state, another update is in progress");
-            return Err(OperationError::InvalidState);
+            debug!("Cancelling abandoned mfareg");
```

Comment:
For clarity we are allowing the regeneration rather than taking by a specific "cancel" action, maybe make this "ignoring abandoned/incomplete"?